src/libmongoc/src/mongoc/mongoc-database.c

```diff
@@ -1100,15 +1098,40 @@ create_collection_with_encryptedFields (mongoc_database_t *database,
    bool state_collections_ok =
       create_encField_state_collection (
          database, encryptedFields, name, "esc", error) &&
-      create_encField_state_collection (
-         database, encryptedFields, name, "ecc", error) &&
       create_encField_state_collection (
          database, encryptedFields, name, "ecoc", error);
    if (!state_collections_ok) {
       // Failed to create one or more state collections
       goto fail;
    }
 
+   // Check the wire version to ensure server is 7.0.0 or newer.
```

Comment:
This wire version check should be performed _before_ creating the QEv2 state collections. I opened [CDRIVER-4653](https://jira.mongodb.org/browse/CDRIVER-4653) to track this.